netman/dbus_test.go

```diff
@@ -0,0 +1,272 @@
+	}
+	fmt.Printf("===== Found devices: %v\n", devices)
+}
+
+func TestGetWifiDevices(t *testing.T) {
+	client := NewClient(&mockObj{})
+	devices := client.GetDevices()
+	wifiDevices := client.GetWifiDevices(devices)
+	found1 := false
+	found2 := false
+	for _, v := range devices {
+		switch v {
+		case "/d/1":
+			found1 = true
+		case "/d/2":
+			found2 = true
+		}
+	}
+	if !found1 || !found2 {
+		t.Errorf("An expected device was not found")
+	}
+	if len(wifiDevices) != 2 {
+		t.Errorf("Two wifi device should have been found but, found: %d", len(wifiDevices))
+	}
+	fmt.Printf("===== Found wifi devices: %v\n", wifiDevices)
+}
+
+func TestGetAPs(t *testing.T) {
+	client := NewClient(&mockObj{})
+	devices := client.GetDevices()
+	ap2device := make(map[string]string)
+	wifiDevices := client.GetWifiDevices(devices)
+	aps := client.GetAccessPoints(wifiDevices, ap2device)
+	if len(aps) != 4 {
+		t.Errorf("4 APs  should have been found, but found: %d", len(aps))
+	}
+	fmt.Printf("===== Found APs: %v\n", aps)
+}
+
+func TestGetSsids(t *testing.T) {
+	client := NewClient(&mockObj{})
+	devices := client.GetDevices()
+	wifiDevices := client.GetWifiDevices(devices)
+	ap2device := make(map[string]string)
+	ssid2ap := make(map[string]string)
+	aps := client.GetAccessPoints(wifiDevices, ap2device)
+	ssids := client.GetSsids(aps, ssid2ap)
+	if len(ssids) != 4 {
+		t.Errorf("4 SSIDs should have been found, but found: %d", len(ssids))
+	}
+	fmt.Printf("===== GetSSIDs (ssid/ap): %v\n", ssids)
+}
+
+func TestSsids(t *testing.T) {
+	client := NewClient(&mockObj{})
+	ssids, _, _ := client.Ssids()
+	if len(ssids) != 4 {
+		t.Errorf("4 SSIDs should have been found, but found: %d", len(ssids))
+	}
+	fmt.Printf("===== Ssids() (ssid/ap): %v\n", ssids)
+}
+
+func TestConnected(t *testing.T) {
+	client := NewClient(&mockObj{})
+	if !client.Connected([]string{"d1"}) {
+		t.Errorf("Should have found connected state, but did not")
+	}
+	if client.Connected([]string{}) {
+		t.Errorf("Should have found no connection since there are no devices, but did not")
+	}
+}
+
+func TestConnectedWifi(t *testing.T) {
+	client := NewClient(&mockObj{})
+	if !client.ConnectedWifi([]string{"d1"}) {
+		t.Errorf("Should have found Wificonnected state, but did not")
+	}
+	if client.ConnectedWifi([]string{}) {
+		t.Errorf("Should have found no connection since there are no devices, but did not")
+	}
+}
+
+func TestiDiscconnectWifi(t *testing.T) {
+	client := NewClient(&mockObj{})
+	res := client.DisconnectWifi([]string{})
+	if res != 0 {
+		t.Errorf("0 Disconnect call expected, but found: %d", res)
+	}
+	res = client.DisconnectWifi([]string{"d1"})
+	if res != 1 {
+		t.Errorf("1 Disconnect call expected, but found: %d", res)
+	}
+}
+
+func TestSetIfaceManaged(t *testing.T) {
+	mock := &mockObj{}
+	client := NewClient(mock)
+	res := client.SetIfaceManaged("notaniface", []string{})
+	if res != "" {
+		t.Errorf("No Disconnect call expected, but found: %d", res)
```

Comment:
Travis complains about %d here is not filled with a decimal but a string. A %v for instance, works.